andela/wg-rogue-one
# #157731611 Creates an exercise information read-only API endpoint

### What does this PR do?
This commit creates a new read-only API endpoint that collects all the information
of an exercise such as muscles, category, language, license, category, equipment, status in one request.
Add a new exercises API endpoint to wger/urls.py.
Add an ExerciseReadOnlyViewSet class to wger/exercises/api/views.py and initialize
it as a ReadOnlyModelViewSet to enable only the GET parameter,
Add ExerciseReadOnlySerializer class to wger/exercises/api/serializers.py to convert
the data into JSON format.

#### Description of Task to be completed?
There should be a new read-only API endpoint that collects all the information of an exercise such as muscles, category, images, etc. This makes it easier for clients to get all the information since only a single request is needed to be done.

#### How should this be manually tested?
1: Clone the application and install the requirements in your virtual environment.
2: Update the database settings in the `settings.py` file with your postgres user, password and database name.
3: Run the application with `python manage.py runserver`.
4: Register and login. 
5: Go to `About this Software` on the menu and select `Rest API` from the dropdown list.
6: Click `Browse the API` and select the link attached to`exercises` from JSON object.
7: You should have a `GET` API endpoint displaying all the exercise information.
8: Make a GET request: 
```
GET http://127.0.0.1:8000/api/v2/exercises/
```
9: Select the `id` of the exercise whose information you want to get and add it to the URL. `e.g.` 
```
GET http://127.0.0.1:8000/api/v2/exercises/174
```
9: The expected response(sample):
```
{
    "id": 387,
    "status": "2",
    "license_author": "Don",
    "license": "Creative Commons Attribution Share Alike 4 (CC-BY-SA 4)",
    "category": "Chest",
    "equipment": [
        "Barbell"
    ],
    "description": "<p>Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum</p>",
    "name": "Military",
    "name_original": "Military",
    "muscles": [
        "Anterior deltoid",
        "Triceps brachii"
    ],
    "muscles_secondary": [],
    "creation_date": "2018-05-31",
    "language": "English (en)",
    "uuid": "bd0bdcca-6d27-4f44-9485-e598b4c55f09",
    "exerciseimage_set": [
        {
            "id": 1,
            "license_author": "Don",
            "status": "2",
            "image": "http://127.0.0.1:8000/media/exercise-images/387/Andela_Uganda_Offsite_400-X2.jpg",
            "is_main": true,
            "license": 2,
            "exercise": 387
        }
    ]
}
```
#### Any background context you want to provide?
This feature was not available.

#### What are the relevant pivotal tracker stories?
#157731611
